ta4j-core/src/test/java/org/ta4j/core/analysis/criteria/LinearTransactionCostCriterionTest.java

```diff
@@ -38,7 +38,7 @@
     private ExternalCriterionTest xls;
 
     public LinearTransactionCostCriterionTest(Function<Number, Num> numFunction) throws Exception {
-        super((params) -> new LinearTransactionCostCriterion((double) params[0], (double) params[1], (double) params[2]),numFunction);
+        super((params) -> new LinearTransactionCostCriterion((double) params[0], (double) params[1], (double) params[2], numFunction), numFunction);
```

Comment:
This is starting to feel hideous.  May need a redesign on how `numFunctions` are handled.  They are now
1) stored as an API default
2) stored in TimeSeries
3) passed explicitly

Maybe 2) and 3) should be removed and only 1) used..?  Regardless it feels like it's getting unwieldy.